jmxtrans-output/jmxtrans-output-core/src/main/java/com/googlecode/jmxtrans/model/output/GraphiteWriter.java

```diff
@@ -103,7 +105,12 @@ public void internalWrite(Server server, Query query, ImmutableList<Result> resu
 							log.debug("Graphite Message: {}", line);
 							writer.write(line);
 						} else {
-							log.warn("Unable to submit non-numeric value to Graphite: [{}] from result [{}]", value, result);
+							if (!didWarnAboutNonNumeric) {
```

Comment:
The circuit should probably be broken by `Query` / result and not by output writer. In this implementation, if we have multiple non numeric results, there will only be a warning for the first one. Also, an OutputWriter should be threadsafe (this is not the case for all output writers, and in this case a race condition is not too bad, still...).

Side note, this is not a circuit braker, but a "only once" warning, which is a good thing!